extensions/podman/packages/extension/src/certificate-detection/certificate-detection-service.ts

```diff
@@ -0,0 +1,243 @@
+    try {
+      const entries = await fs.readdir(dirPath, { withFileTypes: true });
+
+      for (const entry of entries) {
+        const fullPath = path.join(dirPath, entry.name);
+
+        if (this.shouldScanDirectory(entry, depth)) {
+          await this.scanDirectory(fullPath, result, depth + 1);
+        } else if (this.shouldCountCertificate(entry, depth)) {
+          result.certificateCount++;
+        }
+      }
+    } catch (error) {
+      if (error instanceof Error) {
+        const errorInfo: DetectionError = {
+          error: error.message,
+          code: (error as NodeJS.ErrnoException).code,
+        };
+        result.errors?.push(errorInfo);
+      }
+    }
+  }
+
+  private shouldScanDirectory(entry: Dirent, depth: number): boolean {
+    return entry.isDirectory() && !entry.isSymbolicLink() && depth === 0 && this.isValidRegistryDirectory(entry.name);
+  }
+
+  private shouldCountCertificate(entry: Dirent, depth: number): boolean {
+    const isValidFile = entry.isFile() || (entry.isSymbolicLink() && this.config.followSymlinks);
+    return isValidFile && depth >= 1 && this.isCertificateFile(entry.name);
+  }
+
+  private isCertificateFile(filename: string): boolean {
+    const ext = path.extname(filename).toLowerCase();
+
+    if (this.config.excludedExtensions.includes(ext)) {
+      return false;
+    }
+
+    return this.config.validCertExtensions.includes(ext);
+  }
+
+  private isValidRegistryDirectory(directoryName: string): boolean {
+    // According to containers-certs.d(5), directory names should follow hostname:port format
+    // Examples: registry.example.com:5000, localhost:5000, example.com
+
+    // Allow hostname without port (default port assumed)
+    const hostnamePattern = /^[a-zA-Z0-9]([a-zA-Z0-9.-]*[a-zA-Z0-9])?$/;
+    const hostnamePortPattern = /^[a-zA-Z0-9]([a-zA-Z0-9.-]*[a-zA-Z0-9])?:\d+$/;
+
+    return hostnamePattern.test(directoryName) || hostnamePortPattern.test(directoryName);
+  }
+
+  protected async getDirectoriesToScan(): Promise<string[]> {
+    const directories: string[] = [];
+
+    directories.push(...this.config.certDirectories.system);
+
+    const homeDir = os.homedir();
+    const userDirs = this.config.certDirectories.user.map(dir => dir.replace('$HOME', homeDir));
+    directories.push(...userDirs);
+
+    const existingDirs: string[] = [];
+    for (const dir of directories) {
+      try {
+        await fs.access(dir, fs.constants.R_OK);
+        existingDirs.push(dir);
+      } catch (error) {
+        const nodeError = error as NodeJS.ErrnoException;
+        if (nodeError.code && nodeError.code !== 'ENOENT') {
+          console.debug(`Certificate directory not accessible: ${dir} (${nodeError.code})`);
+        }
+      }
+    }
+
+    return existingDirs;
+  }
+
+  private async sendTelemetry(result: CertificateDetectionResult): Promise<void> {
+    if (!this.telemetryLogger) {
+      return;
+    }
```

Comment:
I think we have redundant check at different place. either we check before calling or inside the method but both looks it little bit redundant

Reply:
Yes, you're right, removed redundant check